src/afl-forkserver.c

```diff
@@ -1137,6 +1162,15 @@ fsrv_run_result_t afl_fsrv_run_target(afl_forkserver_t *fsrv, u32 timeout,
 
   /* Report outcome to caller. */
 
+  /* TODO We use SIGTERM here as an indicator of Xen mode,
+     although it's not equivalent! */
+  if (fsrv->kill_signal == SIGTERM && !*stop_soon_p &&
```

Comment:
Pretty sure we can just always do the `last_run_timed_out` check here, and drop it below, any reason not to?
Is there a race condition here the recent code could catch?

Also, we can wrap it in a `unlikely`

Reply:
Ah I guess it is one (usually) unneeded jump? Well we don't save that jump by adding another condition, either :)